### ISSUE TITLE
angular/di rule false positive when function has no dependencies

### ISSUE THREAD

**Issue by leonid-shevtsov** [Gillespie59/eslint-plugin-angular#329]
`angular/di` rule when configured to `array` will cause a false positive when the function does not declare any dependencies.

For example, this syntax causes the check to fail:

```
app.factory('FooBar', function() {
})
```

and the check passes if the function is wrapped into an array:

```
app.factory('FooBar', [function() {
}])
```

Expected behavior: no error (rule doesn't apply when there are no dependencies).

**Comment by tilmanpotthof** [comment]
@leonid-shevtsov Thanks for the report.

**Comment by XavierDupessey** [comment]
The "$inject" syntax is also affected:

This check fails:
```
angular.module('FooBar').config(config);
function config () {}
```

Whereas that one passes:
```
angular.module('FooBar').config(config);
config.$inject = [];
function config () {}
```
